### DISCUSSION TITLE
Premium search pagination unavailable

### DISCUSSION THREAD

**Discussion by xiaolin-TL** [tweepy/tweepy#1424] [Questions] [answered]
Tweepy docs for `API.search_30_day` and `API.search_full_archive` state: 
```
next – This parameter is used to get the next ‘page’ of results. The value used with the parameter is pulled directly from the response provided by the API, and should not be modified.
```
But the `ResultSet` from the two methods does not contain the next parameter (only max_id and since_id). How does pagination work with premium search?

**Comment by Harmon758** [comment, marked as answer]
Tweepy handles pagination for you with Cursor objects. See https://tweepy.readthedocs.io/en/latest/cursor_tutorial.html.
Both `API.search_30_day` and `API.search_full_archive` can be used with them.
If you need the actual `next` token, `NextIterator`, the iterator that would be returned by `Cursor.pages` in this case, has a `next_token` attribute.
